prediction_market_agent/tools/anvil/models.py

```diff
@@ -0,0 +1,98 @@
+import typing as t
+
+from eth_typing import ChecksumAddress
+from prediction_market_agent_tooling.tools.contract import SimpleTreasuryContract
+from prediction_market_agent_tooling.tools.hexbytes_custom import HexBytes
+from pydantic import BaseModel, Field, computed_field
+
+from prediction_market_agent.agents.microchain_agent.nft_treasury_game.deploy_nft_treasury_game import (
+    DEPLOYED_NFT_AGENTS,
+)
+from prediction_market_agent.tools.message_utils import unzip_message_else_do_nothing
+
+
+class EventBase(BaseModel):
+    block_number: int = Field(alias="blockNumber")
+    transaction_hash: HexBytes = Field(alias="transactionHash")
+    address: HexBytes
+    event: str
+
+
+class ERC721Transfer(EventBase):
+    from_address: HexBytes
+    to_address: HexBytes
+    token_id: int
+
+    @classmethod
+    def from_event_log(cls, log: dict[t.Any, t.Any]) -> "ERC721Transfer":
+        d = {
+            "from_address": log["args"]["from"],
+            "to_address": log["args"]["to"],
+            "token_id": log["args"]["tokenId"],
+        }
+        return ERC721Transfer.model_validate({**d, **log})
+
+
+class AgentCommunicationMessage(EventBase):
```

Comment:
We already have `fetch_unseen_transactions` and [MessageContainer](https://github.com/gnosis/prediction-market-agent-tooling/blob/707a76505661165e00864e89791f956531127b08/prediction_market_agent_tooling/tools/data_models.py#L9), is redefining stuff really necessary?

Reply:
I was also doing a bit of crawling of event data (tx hash, etc), but not extremely important for now.
Refactored and now using `MessageContainer`